include/random.h

```diff
@@ -37,6 +37,13 @@ static inline ticks getticks(void)
     __asm__ __volatile__("mrs %0, cntvct_el0" : "=r"(val));
     return val;
 }
+#elif defined(__riscv)
```

Comment:
Shall we check CSR before using `rdtime` pseudo -instruction? e.g.,
```
/* Test for Zicsr extension, if relevant */
#if defined(__riscv_arch_test)
#if !defined(__riscv_zicsr)
#error "-march must include zicsr to access CSRs" 
#endif
```

Reply:
Hi @jserv,
Most Linux-based environments imply the use of the zicsr when building. I believe the checks can be safely ignored.

Reply:
> Hi @jserv, Most Linux-based environments imply the use of the zicsr when building. I believe the checks can be safely ignored.

Mention the assumption in the comments or git commit message.

Reply:
Added comment, also, we need to get upper 32 bits via `timeh` CSR to concatenate a 64 bits time value on RV32.